### PR TITLE
Fix concurrency problem in staking tests affecting CI

### PR DESCRIPTION
**Pull request by klntsky**

### CODE REVIEW

**Review by jy14898 — CHANGES_REQUESTED** [review]
Other than these changes, LGTM

File: test/Plutip/Staking.purs [hunk 1]
```diff
@@ -113,8 +113,7 @@ suite = do
   -- We must never select this pool, because it retires at the third epoch
   -- (this is Plutip internal knowledge)
   -- https://github.com/mlabs-haskell/plutip/blob/7f2d59abd911dd11310404863cdedb2886902ebf/src/Test/Plutip/Internal/Cluster.hs#L692
-  retiringPoolId <- liftEffect
-    $ maybe (throw "unable to decode poolId bech32") pure
+  retiringPoolId <- liftEffect $ liftM (error "unable to decode poolId bech32")
```
Comment (jy14898) [inline test/Plutip/Staking.purs:116]: I don't think this `liftEffect` is necessary, due to `liftM` being over any `m` with `MonadError Error m`

**Review by jy14898 — APPROVED** [review]
LGTM, small comment but not important

**Review by rynoV — APPROVED** [review]

**Review by rynoV — COMMENTED** [review]
Suggestion for changelog:

```
- Module `Contract.Transaction` now exports `mkPoolPubKeyHash` and `poolPubKeyHashToBech32` for bech32 string <-> `PoolPubKeyHash` conversion
```

**Comment by zmrocze** [comment]
The spawn and configuration of this pool is handled at plutip level already: 
https://github.com/mlabs-haskell/plutip/blob/7f2d59abd911dd11310404863cdedb2886902ebf/src/Test/Plutip/Internal/Cluster.hs#L690
so the change could have been made there

**Pull request merged**